### PR TITLE
Update port forwarding and Remote-Tunnels docs

### PR DESCRIPTION
**Pull request by bamurtaugh**
Update our tunneling-related docs:
- Mention Microsoft dev tunnels specifically 
     - cc @curib please let me know if we should use any other capitalization, terminology, docs links, etc!
- Cross-link to info on Dev Containers + Remote-Tunnels (cc @chrmarti)

### CODE REVIEW

File: docs/remote/tunnels.md [hunk 1]
```diff
@@ -164,4 +166,4 @@ As mentioned in the [`code` CLI doc](/docs/editor/command-line.md#create-remote-
 
 ### Can I use other Remote Development Extensions or a dev container while I'm tunneling?
 
-Currently, you can connect to WSL over Remote - Tunnels.
+Yes! Currently, you can connect to [WSL](/docs/remote/wsl.md) and [dev containers](/docs/devcontainers/containers.md#open-a-folder-on-a-remote-tunnel-host-in-a-container) over Remote - Tunnels.
```
Comment (connor4312) [inline docs/remote/tunnels.md:169]: We should mention dev containers now works here too! (Only on vscode desktop right now, vscode.dev support is still on the way)

Comment (bamurtaugh) [reply docs/remote/tunnels.md:169]: Yes agree! Wondering if the current mention of dev containers sounds good, or if you think we should change it?

Comment (connor4312) [reply docs/remote/tunnels.md:169]: Oh, I see we have docs for that, but this heading says "Can I use other Remote Development Extensions or a dev container while I'm tunneling?" and the answer is now 'yes to both'

Comment (bamurtaugh) [reply docs/remote/tunnels.md:169]: Ah, so maybe while my current phrasing here lists WSL and dev containers in one sentence, it'd instead be clearer to separate them into 2 bullet points or spaced out sentences (where we can expand a bit upon each, such as the dev containers web vs desktop limitations you mention)?

Or were you mostly suggesting rephrasing the "Yes" to "Yes to both" to start off the section being more explicit?

Apologies if I'm still missing your main suggestion for this section!

Comment (connor4312) [reply docs/remote/tunnels.md:169]: Oh, my bad, my brain read everything after `[WSL]` to the tunnel as one link and I skipped over that dev containers was mentioned 🤦  lgtm

**Review by connor4312 — APPROVED** [review]

**Comment by bamurtaugh** [comment]
Thanks @connor4312!

@gregvanl this is ready for your review whenever you get a chance!

**Comment by curib** [comment]
This looks good. Thanks for making these changes and tagging me, Brigit!

**Pull request merged**